readthedocs/oauth/migrations/0013_data_migration_for_remote_relation_model.py

```diff
@@ -0,0 +1,49 @@
+# Generated by Django 2.2.16 on 2020-10-05 06:15
+
+import json
+from itertools import islice
+
+from django.db import migrations
+
+
+def move_data_to_remote_relations(apps, schema_editor):
+    RemoteRelation = apps.get_model('oauth', 'RemoteRelation')
+
+    def remote_relations_generator(relations):
+        for relation in relations:
+            relation.account = relation.remoterepository.account
+            relation.active = relation.remoterepository.active
+            relation.admin = relation.remoterepository.admin
+            relation.pub_date = relation.remoterepository.pub_date
+            try:
+                relation.json = json.loads(relation.remoterepository.json)
+            except json.decoder.JSONDecodeError:
+                pass
+
+            yield relation
+
+    relations_queryset = RemoteRelation.objects.all().select_related('remoterepository')
+    remote_relations = remote_relations_generator(relations_queryset)
+    batch_size = 5000
+
+    while True:
+        batch = list(islice(remote_relations, batch_size))
+
+        if not batch:
+            break
+        RemoteRelation.objects.bulk_update(
+            batch,
+            ['account', 'active', 'admin', 'pub_date', 'json'],
+            batch_size
+        )
```

Comment:
Taken from Django's example https://docs.djangoproject.com/en/3.0/ref/models/querysets/#bulk-create

Reply:
Would be good to have it as a comment in the code.

Reply:
Would be good to have it as a comment in the code.